### DISCUSSION TITLE
to_pandas() -> to_df() adding additional columns functionality lost

### DISCUSSION THREAD

**Discussion by DriesDeprest** [PySport/kloppy#203] [Q&A] [answered]
In the [to_pandas()](https://github.com/PySport/kloppy/blob/master/kloppy/domain/models/event.py#L813-L851) method of our EventDataset class, we have an `additional_columns` argument which allows us to add extra columns using e.g. the following approach:
``` python

events_dataset.to_pandas(
        additional_columns={"opponent_team_id": lambda event: int(
            sp_dataset.metadata.teams[1].team_id
            if event.team == sp_dataset.metadata.teams[0]
            else sp_dataset.metadata.teams[0].team_id
        ),
        "home_team_id": lambda event: int(sp_dataset.metadata.teams[0].team_id),
        "away_team_id": lambda event: int(sp_dataset.metadata.teams[1].team_id),
        },
    )
```
However, with the to_pandas() method being deprecated, I wanted to start replacing them to the to_df() method. However, this method does not seem to offer the same functionality to add additional_columns.

How should we go about this?

**Comment by koenvo** [comment, marked as answer]
Would this work for you? https://kloppy.pysport.org/examples/event_data/#Use-metadata-when-transforming-to-pandas-dataframe

```python
events_dataset.to_df(
  "*",
  opponent_team_id=lambda event: int(
      sp_dataset.metadata.teams[1].team_id
      if event.team == sp_dataset.metadata.teams[0]
      else sp_dataset.metadata.teams[0].team_id
  ),
  home_team_id=int(sp_dataset.metadata.teams[0].team_id), # don't need to use lambda as the data is independent of `event`
  away_team_id=int(sp_dataset.metadata.teams[1].team_id) # don't need to use lambda as the data is independent of `event`
)
```

I didn't really know what `sp_dataset` is. You can also use `event.dataset` to refer to the dataset the event is part of.

**Reply by DriesDeprest** [reply]
Yes, this approach works. I struggled with finding out the right syntax.

Sorry for the delay in my response, I missed that you replied.
